controllers/glance_controller.go

```diff
@@ -930,3 +970,47 @@ func (r *GlanceReconciler) registeredLimitsDelete(
 	}
 	return nil
 }
+
+// GlanceAPICleanup - Delete the glanceAPI instance if it no longer appears
+// in the spec.
+func (r *GlanceReconciler) glanceAPICleanup(ctx context.Context, instance *glancev1.Glance) error {
+	// Generate a list of GlanceAPI CRs
+	apis := &glancev1.GlanceAPIList{}
+	listOpts := []client.ListOption{
+		client.InNamespace(instance.Namespace),
+	}
+	if err := r.Client.List(ctx, apis, listOpts...); err != nil {
+		r.Log.Error(err, "Unable to retrieve GlanceAPI CRs %v")
+		return nil
+	}
```

Comment:
That makes sense, but In order to avoid issues to the `glanceAPICleanup` function, another way of doing this is to move it outside of the loop that becomes as follow:

```golang
	for name, glanceAPI := range instance.Spec.GlanceAPIs {
		err = r.apiDeployment(ctx, instance, name, glanceAPI, helper, serviceLabels)
		if err != nil {
			return ctrl.Result{}, err
		}
	}
	err = r.glanceAPICleanup(ctx, instance)
	if err != nil {
		return ctrl.Result{}, err
	}
```
by doing this, every time a Reconcile is triggered (even when I `oc patch glance glance --type=json -p="[{'op': 'remove', 'path': '/spec/glanceAPIs/defaultX'}]"` I can make sure the resources are cleaned up.
How does this solution sound?

Reply:
That looks good to me!